### PR TITLE
fixed problems with the streaming-bigdata application

### PR DESCRIPTION
**Pull request by ShayHassidim**
fixed problems with the streaming-bigdata to match the wiki docs. 
See changes with the common , processor and feeder.

### CODE REVIEW

File: apps/streaming-bigdata/feeder/src/main/java/org/openspaces/bigdata/feeder/TwitterHomeTimelineFeederTask.java [hunk 1]
```diff
@@ -80,6 +84,12 @@ public void run() {
     			}
     			logTweet(publicTweet);
     			gigaSpace.write(buildTweet(publicTweet));
+    			try {
+					Thread.sleep(3000);
+				} catch (InterruptedException e) {
+					// TODO Auto-generated catch block
```
Comment (itaifrenkel) [inline apps/streaming-bigdata/feeder/src/main/java/org/openspaces/bigdata/feeder/TwitterHomeTimelineFeederTask.java:90]: Log the exception.
 rethrow the exception as RuntimeException - The  thread is going down isn't it?

File: apps/streaming-bigdata/feeder/src/main/resources/META-INF/spring/feeder.properties [hunk 2]
```diff
@@ -13,5 +13,5 @@
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *******************************************************************************/
-tweet.delayInMs=60000
```
Comment (itaifrenkel) [inline apps/streaming-bigdata/feeder/src/main/resources/META-INF/spring/feeder.properties:16]: what is the reason for this change? Please add some comment.

File: apps/streaming-bigdata/processor/cassandra-schema.txt [hunk 3]
```diff
@@ -0,0 +1 @@
+create keyspace TWITTER;
```
Comment (itaifrenkel) [inline apps/streaming-bigdata/processor/cassandra-schema.txt:1]: The file moved to apps/streaming-bigdata/bigDataApp/cassandra/cassandra-schema.txt.
This is where the cloudify recipe is. It is not needed here. Please revert this change.

File: apps/streaming-bigdata/processor/src/main/java/org/openspaces/bigdata/processor/FileArchiveOperationHandler.java [hunk 4]
```diff
@@ -95,6 +95,6 @@ private void closeQuietly(Closeable closeable) {
 
 	@Override
 	public boolean supportsBatchArchiving() {
-		return false;
+		return true;
```
Comment (itaifrenkel) [inline apps/streaming-bigdata/processor/src/main/java/org/openspaces/bigdata/processor/FileArchiveOperationHandler.java:98]: The FileArchiveOperationHandler is neither atomic nor idempotent (see http://www.gigaspaces.com/docs/JavaDoc9.1/org/openspaces/archive/ArchiveOperationHandler.html ). Therefore you must return false.

The problem you encountered ClassCastException is in TweetArchiveFilter. It should implement EventExceptionHandler of Object instead of EventExceptionHandler of SpaceDocument[].
And then with reflection determine if it should be casted to "SpaceDocument[]" or "SpaceDocument"

File: apps/streaming-bigdata/processor/src/main/java/org/openspaces/bigdata/processor/ServiceDiscoveryPropertyPlaceHolderConfigurer.java [hunk 5]
```diff
@@ -66,8 +66,10 @@ public void setMinimumNumberOfInstances(int minimumNumberOfInstances) {
 	
 	@Override
 	public void afterPropertiesSet() throws Exception {
-		final FullServiceName fullName = ServiceUtils
-				.getFullServiceName(clusterInfo.getName());
+		FullServiceName fullName = new FullServiceName("twitter-wordcount","twitter-wordcount");
+		if (clusterInfo!=null)
+			fullName = ServiceUtils.getFullServiceName(clusterInfo.getName());
```
Comment (itaifrenkel) [inline apps/streaming-bigdata/processor/src/main/java/org/openspaces/bigdata/processor/ServiceDiscoveryPropertyPlaceHolderConfigurer.java:71]: First, ServiceDiscoveryPropertyPlaceHolderConfigurer is a generic component that should be re-usable and it is not specific to this example.
Second, you can disable it from  processor\src\main\resources\META-INF\spring\pu.xml. It's sole purpose is to inject cassandra.hostname, which you are not using in the first place. I don't get this change. please explain or revert.

File: apps/streaming-bigdata/processor/src/main/resources/META-INF/spring/pu.xml [hunk 6]
```diff
@@ -41,12 +41,14 @@
     <os-core:giga-space id="clusteredGigaSpace" space="space" tx-manager="transactionManager" clustered="true"/>
 
 
-	<!-- fileExternalPersistence is meant for testing purposes, and archives to a file in the local file system 
+	<!-- fileExternalPersistence is meant for testing purposes, and archives to a file in the local file system 	-->
+	
 	<bean id="fileArchiveHandler" class="org.openspaces.bigdata.processor.FileArchiveOperationHandler">
 		<constructor-arg index="0" value="tweetRepo.txt"/>
-	</bean>-->
```
Comment (itaifrenkel) [inline apps/streaming-bigdata/processor/src/main/resources/META-INF/spring/pu.xml:47]: We have decided that cassandra is the default implementation that we ship out with. So we cannot merge this change. If you wish then you can try to use spring profiles, and then inject the profile from cloudify in the context property (I haven't tried it). That way you could control from cloudify the default, without changing pu.xml. Until then, please revert this change.

File: apps/streaming-bigdata/processor/src/main/resources/META-INF/spring/pu.xml [hunk 7]
```diff
@@ -54,24 +56,23 @@
...
 		id="archiveContainer" 
 		giga-space="gigaSpace" 
-		archive-handler="cassandraArchiveHandler"
+		archive-handler="fileArchiveHandler"
 		concurrent-consumers="${archiver.threads}" 
 		max-concurrent-consumers="${archiver.threads}"
-		batch-size="${archiver.batch.size}" >
-		
+		batch-size="100" >
     	<os-archive:tx-support tx-manager="transactionManager"/>
```
Comment (itaifrenkel) [inline apps/streaming-bigdata/processor/src/main/resources/META-INF/spring/pu.xml:73]: why change property with hardocded value?

Comment (ShayHassidim) [inline apps/streaming-bigdata/processor/src/main/resources/META-INF/spring/pu.xml:73]: because it makes users life simple when debugging / testing when running within eclipse - see the wiki content. It is important step before fully deploying into the service grid

File: apps/streaming-bigdata/processor/src/main/resources/META-INF/spring/pu.xml (hunk 6 above)
Comment (ShayHassidim) [inline apps/streaming-bigdata/processor/src/main/resources/META-INF/spring/pu.xml:47]: Having such default makes the user experience very complicated. It means more risk for the user to fail running the tutorial. More moving parts , more risks. The wiki doc also mentioning the fileArchiveHandler is the default. Remember most of the the users are not familiar with big data nor cassandra. Lets make this demo simple and add more complex pieces in a gradual manner..

File: apps/streaming-bigdata/processor/cassandra-schema.txt (hunk 3 above)
Comment (ShayHassidim) [inline apps/streaming-bigdata/processor/cassandra-schema.txt:1]: The wiki content been updated to mention this file located in is current position. Why you need to move it again?

File: apps/streaming-bigdata/feeder/src/main/resources/META-INF/spring/feeder.properties (hunk 2 above)
Comment (ShayHassidim) [inline apps/streaming-bigdata/feeder/src/main/resources/META-INF/spring/feeder.properties:16]: Since this gives the users illusion something is going on frequently. 60 sec delay is too much. This about the user testing this with the list based feeder that is disconnected from the internet.

File: apps/streaming-bigdata/processor/src/main/java/org/openspaces/bigdata/processor/FileArchiveOperationHandler.java (hunk 4 above)
Comment (ShayHassidim) [inline apps/streaming-bigdata/processor/src/main/java/org/openspaces/bigdata/processor/FileArchiveOperationHandler.java:98]: just make sure the FileArchiveOperationHandler works. This fix solved the problem.

File: apps/streaming-bigdata/feeder/src/main/java/org/openspaces/bigdata/feeder/TwitterHomeTimelineFeederTask.java (hunk 1 above)
Comment (ShayHassidim) [inline apps/streaming-bigdata/feeder/src/main/java/org/openspaces/bigdata/feeder/TwitterHomeTimelineFeederTask.java:90]: missed this
